Gems/PhysX/Code/Source/Scene/PhysXScene.cpp

```diff
@@ -1266,4 +1269,20 @@ namespace PhysX
         AZ_PROFILE_DATAPOINT(Physics, stats.nbLostTouches, RootCategory, CollisionsSubCategory, "LostTouches");
         AZ_PROFILE_DATAPOINT(Physics, stats.nbPartitions, RootCategory, CollisionsSubCategory, "Partitions");
     }
+
+    void PhysXScene::SyncActiveBodyTransform(const AzPhysics::SimulatedBodyHandleList& activeBodyHandles)
+    {
+        for (const AzPhysics::SimulatedBodyHandle& bodyHandle : activeBodyHandles)
+        {
+            if (auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get())
```

Comment:
`sceneInterface` could be obtained and checked outside the loop.

Reply:
We could also do an early return too in case it's null and remove some indentation 🙂

Reply:
AZ::Interface<>::Get() is not free - it has to acquire a lock on a mutex to be thread safe. So it's always a good idea to avoid repeated calls in any high frequency code.

You should also update RigidBodyComponent to have a new member variable that is a pointer to the SceneInterface, and get the scene in Activate(), then re-use it. It shows up in profiling because AZ::Interface<AzPhysics::SceneInterface>::Get() is called inside GetRigidBody(), which is called all over the place.

Reply:
yep, made it cached in the component. thanks!